src/repository/transform_photos.py

```diff
@@ -0,0 +1,181 @@
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Photo with id {photo_id} wasn't found",
+        )
+    original_photo.url = photo_url
+    original_photo.updated_by = updated_by
+    original_photo.is_transformed = True
+    original_photo.description = photo_description
+    db.add(original_photo)
+    db.commit()
+    db.refresh(original_photo)
+    return original_photo
+
+
+async def create_transformed_photo_in_db(
+    db: Session,
+    orig_photo_id: int,
+    photo_url: str,
+    updated_by: int,
+    photo_description: str,
+) -> Photo:
+    """
+    Method that creates the new row in the DB table public.photos for transformed photo.
+
+    :param db: DB instance.
+    :type db: Session.
+    :param orig_photo_id: Original photo identifier.
+    :type orig_photo_id: int.
+    :param photo_url: Original photo URL.
+    :type photo_url: str.
+    :param updated_by: User who updated a photo.
+    :type updated_by: int.
+    :param photo_description: Photo description.
+    :type photo_description: str.
+    :return: Photo instance.
+    :rtype: Photo.
+    """
+    new_photo = Photo(
+        url=photo_url,
+        description=photo_description,
+        created_by=updated_by,
+        original_photo_id=orig_photo_id,
+        is_transformed=True,
+    )
+    db.add(new_photo)
+    db.commit()
+    db.refresh(new_photo)
+    return new_photo
+
+
+async def save_transformed_photo_to_db(
+    db: Session,
+    transformed_photo_url: str,
+    updated_by: int,
+    to_override_orig_photo,
+    photo_description: str,
+    orig_photo_id: int,
+) -> Type[Photo] | None | Photo:
+    """
+    Method covers the logic of saving transformed photo in the DB.
+
+    :param db: DB instance.
+    :type db: Session.
+    :param transformed_photo_url: URL of the transformed photo.
+    :type transformed_photo_url: str.
+    :param updated_by: User who updated a photo.
+    :type updated_by: int.
+    :param to_override_orig_photo: Parameter responsible for overriding an original
+    photo or not.
+    :type to_override_orig_photo: bool.
+    :param photo_description: Photo description.
+    :type photo_description: str.
+    :param orig_photo_id: Original photo identifier.
+    :type orig_photo_id: int.
+    :return: Photo instance.
+    :rtype: Type[Photo] | None | Photo.
+    """
+    if to_override_orig_photo:
+        return await update_orig_photo_with_transformed_photo(
+            db=db,
+            photo_id=orig_photo_id,
+            photo_url=transformed_photo_url,
+            updated_by=updated_by,
+            photo_description=photo_description,
+        )
+    else:
+        return await create_transformed_photo_in_db(
+            db=db,
+            orig_photo_id=orig_photo_id,
+            photo_url=transformed_photo_url,
+            updated_by=updated_by,
+            photo_description=photo_description,
+        )
+
+
+async def apply_transformation(
```

Comment:
You can test just this func as a real and only one usage in routes in different cases. Other are technically helpers for this one.